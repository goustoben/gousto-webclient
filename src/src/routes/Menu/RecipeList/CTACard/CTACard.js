import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'

import Recipe from 'containers/menu/Recipe'

const propTypes = {
  isCurrentCollectionRecommendation: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  thematicName: PropTypes.string,
  deliveryDate: PropTypes.string,
  collectionFilterChange: PropTypes.func
}

const CTACard = ({ isCurrentCollectionRecommendation, thematicName, deliveryDate, collectionFilterChange }) => {
  if (thematicName) {
    // if no date is available, add 3 to the current date for the earliest possible delivery date
    const selectedDate = deliveryDate || moment().add(3, 'days').toISOString()

    return (
      <Recipe
        key={'ctaThematic'}
        view={'ctaThematic'}
        thematicName={thematicName}
        selectedDate={selectedDate}
      />
    )
  }

  if (isCurrentCollectionRecommendation) {
    return (
      <Recipe
        key={'ctaAllRecipe'} 
        view={'ctaAllRecipe'} 
        collectionFilterChange={collectionFilterChange}
      />
    )
  }

  return null
}

CTACard.propTypes = propTypes

export { CTACard }
