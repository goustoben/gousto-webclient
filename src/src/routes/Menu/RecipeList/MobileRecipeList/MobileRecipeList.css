.mobileRecipeList {
  display: flex;
  flex-direction: column;
}
