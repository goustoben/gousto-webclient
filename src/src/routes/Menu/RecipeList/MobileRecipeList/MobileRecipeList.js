import React from 'react'
import PropTypes from 'prop-types'
import Immutable from 'immutable'

import { RecipeCard } from '../RecipeCard'
import { CTACard } from '../CTACard'

import css from './MobileRecipeList.css'

const MobileRecipeList = ({
  recipes, mobileGridView, showDetailRecipe, thematicName, isCurrentCollectionRecommendation, deliveryDate, collectionFilterChange
}) => {
  // eslint-disable-next-line react/prop-types
  const createRecipeCard = (value, index) => {
    return (
      <RecipeCard
        key={`${index}-${value.get('id')}`}
        recipe={value}
        index={index}
        mobileGridView={mobileGridView}
        showDetailRecipe={showDetailRecipe}
      />
    )
  }

  return (
    <div className={css.mobileRecipeList}>
      {recipes.map(createRecipeCard)}

      <CTACard
        thematicName={thematicName}
        isCurrentCollectionRecommendation={isCurrentCollectionRecommendation}
        deliveryDate={deliveryDate}
        collectionFilterChange={collectionFilterChange}
      />
    </div>
  )
}

MobileRecipeList.propTypes = {
  mobileGridView: PropTypes.bool,
  showDetailRecipe: PropTypes.func,
  recipes: PropTypes.instanceOf(Immutable.List).isRequired,
  isCurrentCollectionRecommendation: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  collectionFilterChange: PropTypes.func,
  thematicName: PropTypes.string,
  deliveryDate: PropTypes.string
}

export { MobileRecipeList }
