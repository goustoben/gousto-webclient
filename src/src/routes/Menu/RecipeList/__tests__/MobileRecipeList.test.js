import React from 'react'
import { shallow } from 'enzyme'
import Immutable from 'immutable'

import { MobileRecipeList } from '../MobileRecipeList'
import { CTACard } from '../CTACard'

describe('MobileRecipeList', () => {
  describe('when given 4 recipes', () => {
    test('should render correctly', () => {
      const wrapper = shallow(
        <MobileRecipeList
          recipes={Immutable.fromJS([ {}, {}, {}, {} ])}
          mobileGridView={false}
          showDetailRecipe={() => null}
          thematicName={null}
          isCurrentCollectionRecommendation={false}
          deliveryDate={null}
          collectionFilterChange={() => null}
        />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  test('should render a CTACard', () => {
    const wrapper = shallow(
      <MobileRecipeList
        recipes={Immutable.fromJS([])}
      />
    )

    expect(wrapper.find(CTACard)).toHaveLength(1)
  })
})
