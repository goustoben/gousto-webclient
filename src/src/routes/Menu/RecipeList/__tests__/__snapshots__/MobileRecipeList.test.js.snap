// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileRecipeList when given 4 recipes should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MobileRecipeList
    collectionFilterChange={[Function]}
    deliveryDate={null}
    isCurrentCollectionRecommendation={false}
    mobileGridView={false}
    recipes={
      Immutable.List [
        Immutable.Map {},
        Immutable.Map {},
        Immutable.Map {},
        Immutable.Map {},
      ]
    }
    showDetailRecipe={[Function]}
    thematicName={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Immutable.List [
          <Connect(RecipeCard)
            index={0}
            mobileGridView={false}
            recipe={Immutable.Map {}}
            showDetailRecipe={[Function]}
          />,
          <Connect(RecipeCard)
            index={1}
            mobileGridView={false}
            recipe={Immutable.Map {}}
            showDetailRecipe={[Function]}
          />,
          <Connect(RecipeCard)
            index={2}
            mobileGridView={false}
            recipe={Immutable.Map {}}
            showDetailRecipe={[Function]}
          />,
          <Connect(RecipeCard)
            index={3}
            mobileGridView={false}
            recipe={Immutable.Map {}}
            showDetailRecipe={[Function]}
          />,
        ],
        <CTACard
          collectionFilterChange={[Function]}
          deliveryDate={null}
          isCurrentCollectionRecommendation={false}
          thematicName={null}
        />,
      ],
      "className": "mobileRecipeList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0-undefined",
        "nodeType": "class",
        "props": Object {
          "index": 0,
          "mobileGridView": false,
          "recipe": Immutable.Map {},
          "showDetailRecipe": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1-undefined",
        "nodeType": "class",
        "props": Object {
          "index": 1,
          "mobileGridView": false,
          "recipe": Immutable.Map {},
          "showDetailRecipe": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2-undefined",
        "nodeType": "class",
        "props": Object {
          "index": 2,
          "mobileGridView": false,
          "recipe": Immutable.Map {},
          "showDetailRecipe": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3-undefined",
        "nodeType": "class",
        "props": Object {
          "index": 3,
          "mobileGridView": false,
          "recipe": Immutable.Map {},
          "showDetailRecipe": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "collectionFilterChange": [Function],
          "deliveryDate": null,
          "isCurrentCollectionRecommendation": false,
          "thematicName": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Immutable.List [
            <Connect(RecipeCard)
              index={0}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />,
            <Connect(RecipeCard)
              index={1}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />,
            <Connect(RecipeCard)
              index={2}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />,
            <Connect(RecipeCard)
              index={3}
              mobileGridView={false}
              recipe={Immutable.Map {}}
              showDetailRecipe={[Function]}
            />,
          ],
          <CTACard
            collectionFilterChange={[Function]}
            deliveryDate={null}
            isCurrentCollectionRecommendation={false}
            thematicName={null}
          />,
        ],
        "className": "mobileRecipeList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0-undefined",
          "nodeType": "class",
          "props": Object {
            "index": 0,
            "mobileGridView": false,
            "recipe": Immutable.Map {},
            "showDetailRecipe": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1-undefined",
          "nodeType": "class",
          "props": Object {
            "index": 1,
            "mobileGridView": false,
            "recipe": Immutable.Map {},
            "showDetailRecipe": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2-undefined",
          "nodeType": "class",
          "props": Object {
            "index": 2,
            "mobileGridView": false,
            "recipe": Immutable.Map {},
            "showDetailRecipe": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3-undefined",
          "nodeType": "class",
          "props": Object {
            "index": 3,
            "mobileGridView": false,
            "recipe": Immutable.Map {},
            "showDetailRecipe": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "collectionFilterChange": [Function],
            "deliveryDate": null,
            "isCurrentCollectionRecommendation": false,
            "thematicName": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
